projects/js-packages/react-data-sync-client/src/DataSyncHooks.ts

```diff
@@ -0,0 +1,48 @@
+import useSWR from 'swr';
+import useSWRMutation, { SWRMutationConfiguration } from 'swr/mutation';
+import { z } from 'zod';
+import { DataSync } from './DataSync';
+
+export function useDataSync<
+	Schema extends z.ZodSchema,
+	Value extends z.infer< Schema >,
+	Key extends string,
+>( namespace: string, key: Key, schema: Schema ) {
+	type MutationArguments = {
+		arg: Value;
+	};
+	const datasync = new DataSync( namespace, key, schema );
+	const { trigger, isMutating } = useSWRMutation< Value >(
+		key,
+		async ( _, data: MutationArguments ) => {
+			const value = data.arg; // schema.parse( data.arg );
+			const response = await datasync.SET( value );
+			// eslint-disable-next-line no-console
+			console.log( 'Response', response );
```

Comment:
Missed by accident?

Reply:
Yesno - this PR isn't the complete DataSync - it's a prototype implementing a simple component. I need to go through a lot of iteration - I'll need this here for now. But, there's a code freeze issue that we need to talk about. I'll slack you.